gateway/mw_redis_cache_test.go

```diff
@@ -8,62 +8,94 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/TykTechnologies/tyk/config"
 
 	"github.com/TykTechnologies/tyk/apidef"
 	"github.com/TykTechnologies/tyk/test"
 )
 
-func TestRedisCacheMiddleware_WithCompressedResponse(t *testing.T) {
-	const path = "/compressed"
-
+func TestRedisCacheMiddleware(t *testing.T) {
 	conf := func(globalConf *config.Config) {
 		globalConf.AnalyticsConfig.EnableDetailedRecording = true
 	}
 	ts := StartTest(conf)
 	defer ts.Close()
 
+	const compressed = "/compressed"
+	const chunked = "/chunked"
 	createAPI := func(withCache bool) {
 		ts.Gw.BuildAndLoadAPI(func(spec *APISpec) {
 			spec.Proxy.ListenPath = "/"
 			spec.CacheOptions.CacheTimeout = 60
 			spec.CacheOptions.EnableCache = withCache
 			UpdateAPIVersion(spec, "v1", func(v *apidef.VersionInfo) {
-				v.ExtendedPaths.Cached = []string{path}
+				v.ExtendedPaths.Cached = []string{compressed, chunked}
 			})
 		})
 	}
 
-	t.Run("without cache", func(t *testing.T) {
-		createAPI(false)
+	type params struct {
+		path             string
+		bodyMatch        string
+		uncompressed     bool
+		transferEncoding []string
+	}
 
-		ts.Run(t, []test.TestCase{
-			{Path: path, Code: 200, BodyMatch: "This is a compressed response"},
-			{Path: path, Code: 200, BodyMatch: "This is a compressed response"},
-		}...)
-	})
+	check := func(t *testing.T, p params) {
+		subCheck := func(t *testing.T, cachingActive bool, p params) {
```

Comment:
invoke t.Helper() here

Reply:
As the assert functions and `t.Run` calls it, the trace is already printed.